### ISSUE TITLE
Puck3d with custom altitude

### ISSUE THREAD

**Issue by RobertGillett** [mapbox/mapbox-maps-ios#1282]
I'm building an app for pilots that will guide them through waypoints in the sky. Im currently using a CustomLayerHost; however, the animation when changing the coordinates and heading is choppy and doesn't give a great user experience. The animation for the 3dPuck using a custom 3d model along with a custom LocationProvider works great, but I am unable to display with an altitude offset. Is there a workaround for this?

**Comment by ZiZasaurus** [comment]
@RobertGillett, while the ModelLayer is internal, you can still use our lower-level API `setLayerProperties(for layerId: String, properties: [String: Any])` to change the altitude of your model. 

```
try! self.mapView.mapboxMap.style.setLayerProperty(
          for: "puck-model-layer", 
          property: "model-translation", 
          value: [0,0,5000])
```
yields the following: 
![Simulator Screen Shot - iPhone 11 - 2022-04-19 at 17 22 22](https://user-images.githubusercontent.com/44972592/164106883-2c44b39b-8a62-43b7-91a6-ab34ad26ef08.png)

Please be aware that while you can use the internal ModelLayer definition to determine the properties you'd like to change, this definition is subject to change in future iterations of the Maps SDK.

**Comment by RobertGillett** [comment]
Thank You @ZiZasaurus  that was very helpful. The only issue with this method is that the Models are occluded at the horizon when there is a terrain layer is present

https://user-images.githubusercontent.com/43982205/164344579-c29d1177-f887-40d2-b31a-7b19df49db06.mov

In the video I have 2 models, the black airplane is using a CustomLayerHost and the white airplane is using a 3D Puck. Do you have any additional suggestions on how I can best implement this?

**Comment by ZiZasaurus** [comment]
@RobertGillett thank you for following up. I will continue investigating this behavior.

**Comment by ZiZasaurus** [comment]
@RobertGillett this issue will be addressed in v10.6

**Comment by RobertGillett** [comment]
@ZiZasaurus That's great! Thank you
